include/Components/Button.hpp

```diff
@@ -0,0 +1,108 @@
+        public:
+            enum Alignment {
+                Left,
+                Center,
+                Right
+            };
+
+            enum Type {
+                Primary,
+                Secondary,
+                Soft,
+                Danger,
+            };
+
+            enum Size {
+                Small,
+                Normal,
+                Large
+            };
+
+            Button(const char* typeName = "Components::Button", bool initRenderer = true);
+            ~Button();
+
+            typedef std::shared_ptr<Button> Ptr;
+			typedef std::shared_ptr<const Button> ConstPtr;
+
+			static Button::Ptr create() { return std::make_shared<Button>(); }
+			static Button::Ptr copy(Button::ConstPtr widget) { return widget ? std::static_pointer_cast<Button>(widget->clone()) : nullptr; }
+
+            void setImageSize(float size) { this->imageSize = {size, size}; this->setImageScaling(-1); }
+            void setImageSize(const tgui::Vector2f &size) { this->imageSize = size; this->setImageScaling(-1); }
+            tgui::Vector2f getImageSize() const { return this->imageSize; }
+
+            void setDistanceBetweenTextAndImage(float distance) { this->distanceBetweenTextAndImage = distance; this->updateSize(); }
+            float getDistanceBetweenTextAndImage() const { return this->distanceBetweenTextAndImage; }
+
+            void setAlignment(Alignment alignment) { this->alignment = alignment; this->updateAligment(); }
+            Alignment getAlignment() const { return alignment; }
+
+            void setOffset(int offset) { this->offset = offset; this->updateSize(); }
+            int getOffset() const { return this->offset; }
+
+            void setType(Type type) { this->type = type; this->updateStyle(); }
+            Type getType() const { return type; }
+
+            void setGhost(bool ghost) { this->ghost = ghost; this->updateStyle(); }
+            bool isGhost() const { return ghost; }
+
+            void setSize(Size size);
+            Size getSize() const { return size; }
+
+            void canBeActive(bool active, bool automatic = true);
+            bool canBeActive() const { return this->activeSignalId > 0; }
+
+            void setActive(bool active) { this->active = active; this->updateActiveStyle(); }
+            bool isActive() const { return this->canBeActive() ? active : false; }
+
+		protected:
+			tgui::Widget::Ptr clone() const override { return std::make_shared<Button>(*this); }
+
+            void setSize(const tgui::Layout2d &size) override;
+            void updateSize() override;
+            void updateTextPosition() override;
+            void recalculateGlyphSize();
+            void toggleActive() { this->setActive(!this->isActive()); }
+
+        private:
+            void updateAligment();
+            void updateStyle();
+            void updateActiveStyle();
+
+            std::shared_ptr<tgui::priv::dev::ImageComponent> imageComponent;
```

Comment:
Unfortunately, it's the only way to control the button image.
I had talked about it on Discord with the creator of TGui, and he told me it was possible to use it if I knew what I was doing. I looked at the source code carefully, so it shouldn't be a problem.

Otherwise, I would have to completely re-implement the button class, including the different button states and the renderer, which would be a lot more work.